### DISCUSSION TITLE
Could it be possible to format outline log differently

### DISCUSSION THREAD

**Discussion by bmm-alc** [outline/outline#7615] [Ideas]
Hi there,

the log format makes hard activity to see it in a log viewer like loki.
Could it be possible to provide different logformat, [logfmt](https://brandur.org/logfmt) is kinda popular, because it can be readable for human and easily parsed, but some json is good too.

btw I find strange to separate the client request log, and the server response? is it because of the asynchronous nature of nodejs ? This makes difficult to track, would it be possible to have that on one line consolidated and have a field with the response time ? 

Best

**Comment by tommoor** [comment]
The log format is `json` if your environment is set to production.

> I find strange to separate the client request log, and the server response?

This is very normal as requests might be tens of seconds long in the case of, e.g. importing a file.

**Reply by bmm-alc** [reply]
perhaps the log could be formatted differently to be easier to parse,
now we have
```
{"label":"http","level":"info","message":"  --> GET /_health 200 5ms 2b"}
``` 
I wish something there could be something like
```
{"label":"http","level":"info","message":"", "method": "GET", "direction"="input", "path"="/_health", "status"=200, duration_ms="5", size_bytes="2"}
```
perhaps a field that allow to correlate also the input and the output request would be usefull.

best
